### PR TITLE
Plasma rifle rebalance

### PR DESCRIPTION
**Pull request by JohnnyDangerSpace**
## About the PR

Changed it from a stupid gimmick item that is basically an upgrade to the laser rifle (with good rng), to a less punishing gimmick item that is less powerful than the laser rifle. As well, buffed the plasma cutter to actually deal damage to walls and removed most of the wall breaking ability from the rifle.

## Why / Balance

The old one sucked, this one is cool.

## Technical details

Just yaml warrior'd most of it, added the wield comp to the plasma rifle and that's about it for anything complex.

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- tweak: Made the plasma rifle fun

### CODE REVIEW

File: Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs [hunk 1]
```diff
@@ -29,17 +29,26 @@ private void OnShot(EntityUid uid, WeaponRandomExplodeComponent component, ShotA
             if (_random.Prob(component.explosionChance))
             {
                 var intensity = 1;
+                var reduction = 1;
+                if (component.reduction > 1)
+                {
+                    reduction = Convert.ToInt32(component.reduction);
```
Comment (taydeo) [inline Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs:35]: image

Comment (taydeo) [reply Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs:35]: HEY JOHN SPACE, DID YOU KNOW YOU CAN MAKE A VARIABLE EITHER EQUAL SOMETHING, OR NOTHING?

File: Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs [hunk 2]
```diff
@@ -29,17 +29,26 @@ private void OnShot(EntityUid uid, WeaponRandomExplodeComponent component, ShotA
...
                     (EntityUid) uid,
                     typeId: "Default",
                     totalIntensity: intensity,
-                    slope: 3,
-                    maxTileIntensity: 5);
+                    slope: (5 / reduction),
+                    maxTileIntensity: (10 / reduction));
+                if (component.destroygun == true)
+                {
+                    QueueDel(uid);
```
Comment (taydeo) [inline Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs:50]: lol thats actually quite good, Quite funny.

File: Resources/Prototypes/_Goobstation/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml [hunk 3]
```diff
@@ -59,6 +59,8 @@
   - type: WeaponRandomExplode
     explosionChance: 0.16
     multiplyByCharge: 1.0
+    reduction: 2.0
+    destroygun: false
```
Comment (taydeo) [inline Resources/Prototypes/_Goobstation/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml:63]: follow naming convention pt2

File: Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeComponent.cs [hunk 4]
```diff
@@ -12,4 +12,16 @@ public sealed partial class WeaponRandomExplodeComponent : Component
...
+    /// decreases the self damage and explosion radius
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public float reduction;
+
+    /// <summary>
+    /// deletes the gun after the explosion if this is true
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public bool destroygun;
```
Comment (taydeo) [inline Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeComponent.cs:26]: naming convention for both of these

```csharp
public float? reduction;

public bool destroyGun = false;
```

**Review by taydeo — CHANGES_REQUESTED** [review]

File: Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs (hunk 1 above)
Comment (taydeo) [reply Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs:35]: also what is this supposed to be doing, you can just get it directly from the component and if you need to convert to an int you can just cast it

**Comment by taydeo** [comment]
before john space gets angry at me i hold myself to these standards too :godo:

File: Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs (hunk 1 above)
Comment (JohnnyDangerSpace) [reply Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs:35]: i will eat you

Comment (Komonighub) [reply Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs:35]: lol lmao

File: Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs [hunk 5]
```diff
@@ -29,18 +29,26 @@ private void OnShot(EntityUid uid, WeaponRandomExplodeComponent component, ShotA
...
                 _explosionSystem.QueueExplosion(
                     (EntityUid) uid,
                     typeId: "Default",
                     totalIntensity: intensity,
-                    slope: 5,
-                    maxTileIntensity: 10);
-                QueueDel(uid);
+                    slope: (5 / reduction),
+                    maxTileIntensity: (10 / reduction));
+                if (component.destroyGun == true) //#funkystation part of comp or something idk go away tay i dont wanna do this shit
```
Comment (taydeo) [inline Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs:48]: redundant check

Comment (JohnnyDangerSpace) [reply Content.Server/_Goobstation/WeaponRandomExplode/WeaponRandomExplodeSystem.cs:48]: it checks if its true and deletes the gun if its true how is that redundant fatso

**Comment by taydeo** [comment]
Does this still crash the game?

**Comment by JohnnyDangerSpace** [comment]
only in testing, once again that isnt an issue with my pr here it's literally an issue with the rifle and always has been one even before this.

**Pull request merged**